DependencyInjection/Security/Factory/OAuthFactory.php

```diff
@@ -31,12 +31,12 @@ public function create(ContainerBuilder $container, $id, $config, $userProvider,
     {
         $providerId = 'security.authentication.provider.fos_oauth_server.'.$id;
         $container
-            ->setDefinition($providerId, new DefinitionDecorator('fos_oauth_server.security.authentication.provider'))
+            ->setDefinition($providerId, new ChildDefinition('fos_oauth_server.security.authentication.provider'))
```

Comment:
Is this compatible with Symfony 2.x? If not, it's a breaking change.

Reply:
i really don't know, in branch 2.8, there ist class 
Symfony\Component\DependencyInjection\ChildDefinition
But it is not compatible with Symfony 4.0, becouse in branch 4.0 there is in class
Symfony\Component\DependencyInjection\DefinitionDecorator

@trigger_error('The '.__NAMESPACE__.'\DefinitionDecorator class is deprecated since Symfony 3.3 and will be removed in 4.0. Use the Symfony\Component\DependencyInjection\ChildDefinition class instead.', E_USER_DEPRECATED);

Reply:
Yes, so we should go with

```php
if (class_exists('DefinitionDecorator')) {
    // Symfony < 4.x, use it
} else {
    // Symfony >= 4.x, use ChildDefinition
}
```

This should work in all cases.

Reply:
ok i can put it there